File: FibonnaciSequence/fibonnaci.rb
#!/bin/env ruby

def fibonnaci( value )
  sequence = []

  ( 0 .. value ).each do | number |
    if number <= 1
      sequence.push( number )
    else
      sequence.push( sequence[ -1 ] + sequence[ -2 ] )
    end
  end

  sequence
end

if __FILE__ == $0

  LIMIT = 44

  sequence = []
  for i in 0 .. 25_000
    # sequence += fibonnaci( LIMIT )
    sequence.push( *( fibonnaci( LIMIT ) ) )
  end

  sequence.each_with_index do | value, index |
    puts '%02d: %9d' % [ index % ( LIMIT + 1 ), value ]
  end

end

File: FibonnaciSequence/fibonnaci.pl
#!/bin/env perl

use strict;
use warnings;

sub fibonnaci {
  my $value = shift;
  
  my @sequence = ();

  for ( my $number = 0; $number <= $value; $number++ ) {
    if ( $number <= 1 ) {
      push ( @sequence, $number );
    } else {
      push ( @sequence, $sequence[ -1 ] + $sequence[ -2 ] );
    }
  }

  return @sequence;
}

if ( ! caller ) {
  my $LIMIT = 44;

  my @sequence = ();
  for ( my $i = 0; $i <= 25000; $i++ ) {
    push ( @sequence, fibonnaci( $LIMIT ) );
  }

  my $index = 0;
  foreach my $value ( @sequence ) {
    printf ( "%02d: %9d\n", ( $index % ( $LIMIT + 1 ), $value ) );
    $index++;
  }
}
